tools/statestore-cli/Dockerfile

```diff
@@ -0,0 +1,17 @@
+# RUST_VERSION should be the value of "channel" in rust-toolchain.toml
+ARG RUST_VERSION=1.85
+FROM mcr.microsoft.com/azurelinux/base/rust:$RUST_VERSION AS build
+
+# Move source code from build context into /app folder
+COPY --link / /app
+WORKDIR /app
+
+# Install additional deps
+RUN tdnf install -y openssl-devel clang
+
+RUN cargo build -p "statestore-cli" --release
```

Comment:
Any reason why we can't copy the built application in?

Reply:
It's a good practice to keep the container sterilized. In this container, we build then copy the bin to the final image to be used.